src/lib/db/user-store.ts

```diff
@@ -133,6 +126,11 @@ class UserStore implements IUserStore {
     }
 
     async getAll(): Promise<User[]> {
+        const users = await this.activeAll().select(USER_COLUMNS);
```

Comment:
Should we guard this change behind a flag?

Reply:
Well, my reasoning is that if there are no service accounts it shouldn't matter, and there should only be service accounts if the flag is enabled.

With this PR, the only place this method is called from is `getBaseUsersAndGroups` in `user-admin` which fetches access options and should encompass both account types:
https://github.com/Unleash/unleash/blob/5ff9fa45e582a3bc1a1132c9ff751935d5ae5363/src/lib/routes/admin-api/user-admin.ts#L313

Besides, https://github.com/Unleash/unleash/pull/2918 reverts this back by making the `user` entity only handle "users" instead of all account types. It changes `getBaseUsersAndGroups` to use the new `accountService.getAll()` method instead.

Reply:
Yeah, it makes sense, it's controlled by the ability to add new service accounts. Fair enough :+1: